linkerd/tls/src/server/mod.rs

```diff
@@ -373,3 +365,38 @@ impl fmt::Display for NoServerTls {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use io::AsyncWriteExt;
+
+    use super::*;
+    use std::str::FromStr;
+
+    #[tokio::test]
+    async fn detect_buffered() {
+        let _ = tracing_subscriber::fmt::try_init();
+
+        let (mut client_io, server_io) = tokio::io::duplex(1024);
+        let input = include_bytes!("testdata/curl-example-com-client-hello.bin");
+        let len = input.len();
+        tokio::spawn(async move {
+            client_io
+                .write_all(&*input)
+                .await
+                .expect("Write must suceed");
```

Comment:
if this *does* fail, the panic won't be propagated, since we `spawn` this, and never await the returned `JoinHandle`. i dunno if that's a big deal, since this should actually never fail, but still, may be worth awaiting the `JoinHandle` at the end of the test to be sure this didn't panic?